src/peer_connection.cpp

```diff
@@ -5510,15 +5510,18 @@ namespace libtorrent
 		{
 			std::vector<boost::asio::mutable_buffer> vec;
 			m_send_buffer.build_mutable_iovec(m_send_buffer.size(), vec);
-			int next_barrier = hit_send_barrier(vec);
-			for (std::vector<boost::asio::mutable_buffer>::reverse_iterator i = vec.rbegin();
-				i != vec.rend(); ++i)
+			int next_barrier;
+			std::vector<boost::asio::const_buffer> inject_vec;
+			std::tie(next_barrier, inject_vec) = hit_send_barrier(vec);
+			for (auto i = inject_vec.rbegin(); i != inject_vec.rend(); ++i)
```

Comment:
I believe the semantics are the same here as before, but I don't believe there's a test, so i can't say for sure. @ssiloti does it look OK to you? Does this interface still support what you need in the reputation plugin?